apps/emqx_bridge/src/emqx_bridge_api.erl

```diff
@@ -485,57 +485,22 @@ schema("/bridges_probe") ->
     RequestMeta = #{module => ?MODULE, method => post, path => "/bridges_probe"},
     case emqx_dashboard_swagger:filter_check_request_and_translate_body(Request, RequestMeta) of
         {ok, #{body := #{<<"type">> := ConnType} = Params}} ->
-            case do_probe(ConnType, maps:remove(<<"type">>, Params)) of
+            case emqx_bridge_resource:create_dry_run(ConnType, maps:remove(<<"type">>, Params)) of
                 ok ->
                     {204};
                 {error, Error} ->
-                    {400, error_msg('TEST_FAILED', Error)}
+                    {400, error_msg('TEST_FAILED', try_json_encode(Error))}
```

Comment:
This might be very nit-picky but I'm thinking about the response code for the error case. When reading here https://developer.mozilla.org/en-US/docs/Web/HTTP/Status it seems like 4xx error codes are supposed to indicate client errors (which I interpret as the client has done something it is not really supposed to do). But this error case happens when the health check of the resource fails which I don't think is a client error. However, this response code was used before the PR as well so I guess this is already a part of the API so feel free to just ignore this comment.